scenes/game/game.gd

```diff
@@ -1312,6 +1326,23 @@ func _on_main_step_action_chosen(choice_index):
 	var chosen_action_type = chosen_action["action_type"]
 	var valid_actions = _get_main_step_actions_of_type(chosen_action_type)
 	match chosen_action_type:
+		Enums.GameAction_MainStepAttachedAction:
+			# filter the valid_actions for effect ids, then map it to the owning holomem
```

Comment:
If you do the thing with just using the index, you won't have to do any funky mapping. Just send up the index of the special action they selected.